calibrators/docs/tutorials/tag_based_pnp_calibrator.md

```diff
@@ -0,0 +1,157 @@
+
+## Environment preparation
+
+### Overall calibration environment
+
+The required space for calibration depends on the vehicle and sensors used. During calibration, please always make sure that tags are detected by both camera and lidar. The user can always check whether the tags appear in the `rviz` and the `image view` UI. For instance, he can observe the cyan rhombus around the tag in `rviz` if the tag is successfully detected by lidar, as shown in the image below. For the camera, if the tag is visible in the `image view` UI, it should be detectable. So far, in our use cases, the tag is always detected as long as it is completely visible in the image.
+
+<p align="center">
+    <img src="../images/tag_based_pnp_calibrator/lidar_detection.jpg"  alt="lidar_detection">
+</p>
+
+### Apriltag (Lidartag)
+
+Details can be found in the [base documentation](../../tag_based_pnp_calibrator/README.md)
+
+## Launching the tool
+
+In this tutorial, we take the X2 project as an example.
+First, run the sensor calibration manager:
+
+```text
+ros2 run sensor_calibration_manager sensor_calibration_manager
+```
+
+In `project`, select `x2`, and in `calibrator`, select `tag_based_pnp_calibrator`. Then, press `Continue`.
+
+<p align="center">
+    <img src="../images/tag_based_pnp_calibrator/menu1.jpg"  alt="menu1">
+</p>
+
+A menu titled `Launcher configuration` should appear in the UI, and the user may change any parameter he deems convenient.
+For this tutorial, we modify the default values `calibration_pairs` from `9` to `8` as the rosbag has 8 apriltag detections and also modify the `camera_name` from `camera0` to `camera6`. After configuring the parameters, click `Launch`.
+
+<p align="center">
+    <img src="../images/tag_based_pnp_calibrator/menu2.jpg"  alt="menu2">
+</p>
+
+The following UI should be displayed. When the `Calibrate` button becomes available, click it.
+If it does not become available, it means that either the required `tf` or services are not available. In this case, since the `tf` are published by the provided rosbag, run the bag with the command `ros2 bag play camera_lidar.db3 --clock -r 0.5` and click the `Calibrate` button.
+
+<p align="center">
+    <img src="../images/tag_based_pnp_calibrator/menu3.jpg"  alt="menu3">
+</p>
+
+## Calibration
+
+The calibration starts automatically after clicking the `Calibrate` button. It will keep calibrating the lidartag detections and apriltag detections until the number of the detections fits the user-defined `calibration_pairs` in the `Launcher configuration`.
+
+When the user starts the calibration, `rviz` and the `image view` should be displayed like below.
+
+<p align="center">
+    <img src="../images/tag_based_pnp_calibrator/visualization1.jpg"  alt="visualization1">
+</p>
+
+After the tools detect the lidartag and apriltag, it will show the detection markers on the `rviz` and the `image view`. The text in rviz will also display the number of pairs of lidartag/apriltag detections.
```

Comment:
```suggestion
After the tools detects lidartags and apriltags, it will show the detection markers on `rviz` and the `image view` respectively. The text in `rviz` will also display the number of pairs of converged lidartag/apriltag detections so far.
```